### DISCUSSION TITLE
Does WSLg System Distro container work on Hyper-v VMs?

### DISCUSSION THREAD

**Discussion by yongyanglai** [microsoft/wslg#92] [General]
**Describe the solution you'd like**
Since RDP connection is built over HvSock and client apps communicate with components in WSLg System Distro container via socket, I guess it may be possible to make WSLg System Distro container work with Linux VMs on Hyper-v? Would you please provide more detailed instructions?

**Comment by spronovo** [comment]
Moving to discussion.

**Comment by spronovo** [comment]
The short answer is that yeah, you can run the WSLg system distro in an Hyper-V VM and create a similar experience than WSL/WSLg offers. In that case you probably wouldn't want to consume the vhd directly... instead if you look at the build process for WSLg you'll see that we first generate a docker container... we extract it's content, than convert to a vhd. In a scenario like this you would probably be best to take the docker container and run that in your VM... you would need to do some patching to get the various socket shared back, etc... but it should be possible to get it to work. Internally, we've run similar configuration on a Pi and remote this over the network back to our PC... it's one way to get Wayland network transparency :-). We don't have a simple set of instruction/recipe to do this and it's a bit tricky to get all the pieces lined up,  but feel free to experiment. Maybe if there is enough interest on this we create a container and recipe configure for something like this eventually.

**Reply by yulongbai-nov** [reply]
Hihi, as of 2025, any documents updated for paring WSLg System Distro with a Hyper-V VMs?

**Comment by timriker** [comment]
This sounds interesting. With a full Hyper-V instance One should be able to add a bridged network adapter while keeping the NAT adapter and get IPv6 native working inside the VM. I tried adding a WSL adapter in my Hyper-V instance, but it does not get an IP. Statically assigning one works, but I suppose if WSL changes networks, that will break again. I'd like to see the WSLg instructions for inside a Hyper-V instance documented someplace.
